modules/nextflow/src/main/groovy/nextflow/script/ProcessEntryHandler.groovy

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2013-2024, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.script
+
+import groovy.util.logging.Slf4j
+import nextflow.Session
+import nextflow.Nextflow
+
+/**
+ * Helper class for process entry execution feature.
+ * 
+ * This feature enables direct execution of Nextflow processes without explicit workflows:
+ * - Single process scripts run automatically: `nextflow run script.nf --param value`
+ * - Multi-process scripts use entry selection: `nextflow run script.nf -entry process:name --param value`
```

Comment:
The `-entry` option is deprecated since 24.10: https://nextflow.io/docs/latest/reference/cli.html#run

I would either extend the script name to include a process name (e.g. `script.nf:name`) or use a different command-line option (e.g. `-process name`).